### PR TITLE
166152318 cache nutritional values

### PR DESCRIPTION
**Pull request by Tevinthuku**
#### Title

Cache nutritional values for a plan

#### Description

The nutritional information should be cached so as to reduce the number of db queries that may slow down the rendering process. The cache should be deleted once a nutrition plan, meal or meal item is added, deleted or edited.

The following pages that are affected by the new change for the better since they will be retrieving the nutrition data from the cache once a plan is viewed.

`http://localhost:8000/en/nutrition/<nutritionId>/view`

`http://localhost:8000/en/user/<userId>/overview`

#### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue) 

- [x] New feature (non-breaking change which adds functionality) 

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) 

- [ ] This change requires a documentation update

#### How Has This Been Tested?
- [x] Unit tests

#### Checklist:

 - [x] - create signals for cache deletion
           
  - [x] - create functionality to save nutritional info to cache
           
   - [x] - create tests for saving info to cache

#### PT stories
[#166152318](https://www.pivotaltracker.com/n/projects/2350046/stories/166152318)

### CODE REVIEW

File: wger/nutrition/models.py [hunk 1]
```diff
@@ -40,6 +40,8 @@
 from wger.utils.models import AbstractLicenseModel
 from wger.utils.units import AbstractWeight
 from wger.weight.models import WeightEntry
+from wger.utils.cache import reset_nutritional_values
```
[the remark this answers is not in the document]
Comment (Tevinthuku) [reply wger/nutrition/models.py]: I have removed the import 😄

**Review by martinMutuma — APPROVED** [review]
Awesome work. I like the speed at which you implemented the feature.
It is also prudent that you only delete from the cache the entry that has been deleted by the user instead of destroying all cache.

**Review by Kimaiyo077 — APPROVED** [review]
Good work Tevin. Your implementation will help in speeding up the application now that all the calculations don't have to be redone every time.

**Review by misocho — APPROVED** [review]
Awesome job implementing the cache nutritional values feature. From the code, we can see that the cache values are being added and deleted.

**Review by justMuriithi — APPROVED** [review]
A superb job doing this with great speed. The cache of nutritional values introduces an element of speed that adds to improved user experience.

**Review by michael-basweti — APPROVED** [review]
Works as the task required

**Pull request merged**